gramps/gui/viewmanager.py

```diff
@@ -580,10 +580,47 @@ def no_del_event(self, *obj):
         hits 'x' multiple times. """
         return True
 
+    def checkdisk(self, dbfolder): # TODO check this function
+        """
+        Check disk space
+        """
+        if win():                               # for windows:
+        # import psutil
+        # DISK = "C:"
+        # free = psutil.disk_usage(DISK).free/(1024*1024)
+        # print(f"{free:.4} Mb free on disk {DISK}")
+            pass
+        else:                                   # for linux:
+            fd = os.open(dbfolder, os.O_RDONLY)
+            stats = os.fstatvfs(fd)
+            os.close(fd)
+            freedisk = round(stats[1] * stats[4]/(1024*1024))
+            return freedisk
+
     def quit(self, *obj):
         """
         Closes out the program, backing up data
         """
+#****************
+        # We must check disk space before:
+        #                   - run (Gramps need space for load config file)
+        #                   - quit (Gramps need some space for:
+        #                                                   - db.commit and close database,
+        #                                                   - save config file)
+        #                   - every backup (we need space 'last_backup_file_size'+ 5%)
+        diskspace = 20             # Disk space for Linux - 20 MB, for Gramps - 5 MB, for backup 'last_backup_file_size' + 5%
+        if (self.checkdisk(config.get('database.path')) < diskspace):
+            self.uistate.push_message(self.dbstate, _("Disk space < " + str(diskspace) + " MB. Quit impossible. Clear your disk and repeat this operation."))
```

Comment:
Use format strings instead of concatenation.

Reply:
I'm not Python man.

Reply:
I rewrite this code. New PR have normal string concatenation.
I check code, all translated strings are present in .po file.